### ISSUE TITLE
Bazel crash (no error) with env HOME=not/a/directory

### ISSUE THREAD

**Issue by c-parsons** [bazelbuild/bazel#16326]
`HOME=not/a/directory bazel --output_user_root=bogus info` used to succeed in giving informative output.
However, as of a few weeks ago, this causes Bazel server to crash with **no error**.

```
$ HOME=not/a/directory bazel --output_user_root=bogus info
Starting local Bazel server and connecting to it...
<server dies, returning failure>
$
```

While this seems like a convoluted test case, this is actually something which hit the effort to move Bazel into AOSP. (We were incorrectly specifying `HOME` environment, alongside other flags, and Bazel used to not care. But now Bazel crashes!)

According to `git bisect`, this breakage was caused by https://github.com/bazelbuild/bazel/commit/86451bb2cfd360dc743cd0c2839063459c2600f4, which I believe you were a reviewer for, @Wyverald.

**Comment by c-parsons** [comment]
(Also re: reproducing this error, the `--output_user_root=bogus` is also important, because otherwise bazel crashes with a descriptive error. The error is only suppressed if both `HOME` and `--output_user_root` are specified)

**Comment by Wyverald** [comment]
Is this a P1? Can you specify a good HOME env var?

**Comment by c-parsons** [comment]
It should be P1 because this is a recent regression that results in a crash bug.

IIRC we have a Bazel policy that crash bugs are P1 by default, and this is a pretty difficult-to-diagnose server crash.

In terms of the effort to move Bazel into AOSP, we can indeed fix our HOME path, but that doesn't change the argument for making this P1.

**Comment by Wyverald** [comment]
Not that I don't believe you, but if you could link to the policy, that'd be great.

I was hoping this could be downgraded since it appears hard to trigger and we already have a crapton of P1 issues to fix before the 6.0 cut. Meanwhile, I'll keep this one in mind.

**Comment by c-parsons** [comment]
No, you're right to question me making an assertion about policy with "IIRC".
Frankly, I can't find the policy, so I'm not sure if I accidentally made it up. If so, my bad.

I still think we should make a case to have this be a P1, because a server crash *with no error info* is not something I think we want in our Bazel 6.0 release.  FYI @lberki

**Comment by lberki** [comment]
@c-parsons emotionally, I feel similarly, but I think a better policy is that a _new_ server crash with no error info should not be in Bazel 6.0; the distinction is important because the opposite would imply that we have to fix all such crashes before Bazel 6.0 can be released.

So I'll thereby add this bug to the hotlist of release blockers unless  @Wyverald has some really great arguments why that should not be the case.
